### DISCUSSION TITLE
Interoperatibility with ROS, FPrime, others

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [nasa/cFS#730] [General]
Hi everyone!

I'm reaching out to the cFS community because my team is looking into interoperability between cFS, ROS, Fprime and other software frameworks that people use to create applications that may be deployed on the Lunar surface. This is part of a new, cross-center project.

If you are working on similar efforts, please reply with some information about your project, toolkit, etc. It doesn't have to be restricted to those three frameworks: anything that tries to address similar concerns, or that would have to interoperate with them, is also of interest.

If you are developing applications that could leverage such facilities for interoperability, please also reach out. If you are doing so as part of a commercial effort, please also indicate your company so that we know which industrial partners are interested.

Thanks!

**Comment by skliper** [comment]
cFS has a few options for interoperability (from the cFS readme):

Other Interfaces:
SIL: Simulink Interface Layer at https://github.com/nasa/SIL
ECI: External Code Interface at https://github.com/nasa/ECI
SBN-Client: External code interface to SBN at https://github.com/nasa/SBN-Client

I think there was at least a design concept related to interfacing ROS to cFS on a recent project.  Not sure how far it got into implementation though.  Hopefully someone w/ better information can fill us in!

**Reply by ivanperez-keera** [reply]
Thanks!

**Comment by thnkslprpt** [comment]
@ivanperez-keera did you see these talks at the Flight Software Workshop?
They might give some ideas of people to contact...
[FSW 2022: The Wrong Way to Integrate cFS and ROS](https://youtu.be/Gvjxwtf6WMo)
[FSW 2022: ROS and cFS System (RACS) - Hiroki Kato](https://youtu.be/A8y0ofD7F-g)
[FSW 2022: Overview of the Space Robot Operating System (Space ROS) - Amalaye Oyake](https://youtu.be/CxG1bXvyyZM)

**Reply by ivanperez-keera** [reply]
Thanks!

**Comment by ivanperez-keera** [comment]
Apart from ROS and Fprime, is there any other flight software, robotics or IoT frameworks that cFS has been interconnected with? Anything else I may be missing?
